src/go/k8s/pkg/resources/superusers.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package resources
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	redpandav1alpha1 "github.com/vectorizedio/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/api/admin"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const scramPandaproxyUsername = "pandaproxy_client"
+
+var _ Resource = &SuperUsersResource{}
+
+// SuperUsersResource is part of the reconciliation of redpanda.vectorized.io CRD
+// focusing on the super users for Schema Registry and Panda proxy
+type SuperUsersResource struct {
+	k8sclient.Client
+	scheme                  *runtime.Scheme
+	pandaCluster            *redpandav1alpha1.Cluster
+	fqdn                    string
+	setSuperUsersInRedpanda bool
+	logger                  logr.Logger
+}
+
+// NewSuperUsers creates SuperUsersResource that managed super users
+// for Schema Registry and Panda proxy
+func NewSuperUsers(
+	client k8sclient.Client,
+	pandaCluster *redpandav1alpha1.Cluster,
+	scheme *runtime.Scheme,
+	fqdn string,
+	setSuperUsersInRedpanda bool,
+	logger logr.Logger,
+) *SuperUsersResource {
+	return &SuperUsersResource{
+		client,
+		scheme,
+		pandaCluster,
+		fqdn,
+		setSuperUsersInRedpanda,
+		logger.WithValues(
+			"Kind", ingressKind(),
+		),
+	}
+}
+
+// Ensure will manage Super users for Panda proxy and Schema Registry for redpanda.vectorized.io custom resource
+func (r *SuperUsersResource) Ensure(ctx context.Context) error {
+	adminInternal := r.pandaCluster.AdminAPIInternal()
+	if adminInternal == nil {
+		return nil
```

Comment:
shouldn't this return error? 🤔

Reply:
I'm not sure. Can Redpanda live without Admin API. I think it can. @dotnwat @mmaslankaprv?

Reply:
oh I see, I would really love for us to test that everything will work in that case for the operator :D of course not in this PR